opa/pkg/main.go

```diff
@@ -0,0 +1,324 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/restmapper"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const outputFile = "/output/opa.json"
+
+var instanceGvr = schema.GroupVersionResource{Group: "insights.fairwinds.com", Version: "v1beta1", Resource: "customcheckinstances"}
+var checkGvr = schema.GroupVersionResource{Group: "insights.fairwinds.com", Version: "v1beta1", Resource: "customchecks"}
+
+func main() {
+	ctx := context.Background()
+	config, err := ctrl.GetConfig()
+	if err != nil {
+		panic(err)
+	}
+	dynamicInterface, err := dynamic.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+	kube, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+	groupResources, err := restmapper.GetAPIGroupResources(kube.Discovery())
+	if err != nil {
+		panic(err)
+	}
+	restMapper := restmapper.NewDiscoveryRESTMapper(groupResources)
+	// TODO filter by namespace
+	checkInstances, err := dynamicInterface.Resource(instanceGvr).Namespace("").List(ctx, metav1.ListOptions{})
+
+	if err != nil {
+		panic(err)
+	}
+	actionItems, err := processAllChecks(ctx, checkInstances.Items, restMapper, dynamicInterface)
+
+	if err != nil {
+		panic(err)
+	}
+	outputFormat := Output{ActionItems: actionItems}
+	value, err := json.Marshal(outputFormat)
+	if err != nil {
+		panic(err)
+	}
+	err = ioutil.WriteFile(outputFile, value, 0644)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func processAllChecks(ctx context.Context, checkInstances []unstructured.Unstructured, restMapper meta.RESTMapper, dynamicInterface dynamic.Interface) ([]ActionItem, error) {
+
+	actionItems := make([]ActionItem, 0)
+
+	for _, checkInstance := range checkInstances {
+		var checkInstanceObject customCheckInstance
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(checkInstance.Object, &checkInstanceObject)
+		if err != nil {
+			return nil, err
+		}
+		logrus.Infof("Starting to process check: %s", checkInstanceObject.Name)
+		check, err := dynamicInterface.Resource(checkGvr).Namespace(checkInstanceObject.Namespace).Get(ctx, checkInstanceObject.Spec.CustomCheckName, metav1.GetOptions{})
+		if err != nil {
+			actionItems = append(actionItems, ActionItem{
+				Title:       checkInstanceObject.Name,
+				Description: "An error occured retrieving the Custom Check for this instance",
+				Remediation: "Make sure that the custom check exists and it is in the same namespace as this instance.",
+				Severity:    0.4,
+				Category:    "Reliability",
+			})
+			logrus.Warnf("Error while retrieving check %+v", err)
+			continue
+		}
+		var checkObject customCheck
+		err = runtime.DefaultUnstructuredConverter.FromUnstructured(check.Object, &checkObject)
+		if err != nil {
+			return nil, err
+		}
+
+		newItems, err := processCheck(ctx, checkObject, checkInstanceObject, restMapper, dynamicInterface)
+		if err != nil {
```

Comment:
looks like we're hitting the k8s API in `processCheck` - probably best to `continue` here

Reply:
At this point there's only a couple spots left that the errors are actually floated up. It feels appropriate to me if the API server is temporarily down, or rejects the request or something, to fail the job. Maybe that's not right though....

Reply:
API server being down definitely deserves an error. I was thinking about RBAC issues, but I defer